pkg/core/region.go

```diff
@@ -761,6 +766,13 @@ func GenerateRegionGuideFunc(enableLog bool) RegionGuideFunc {
 					region.GetReplicationStatus().GetStateId() != origin.GetReplicationStatus().GetStateId()) {
 				saveCache = true
 			}
+			// Do not save to kv, because 1) flashback will be eventually set to
+			// false, 2) flashback changes almost all regions in a cluster.
+			// Saving kv may downgrade PD performance when there are many regions.
+			if region.IsFlashbackChanged(origin) {
+				saveCache = true
+				return
+			}
```

Comment:
I'm not sure if the `saveCache = true` which in `GetReplicationStatus` needs to be returned directly.
And is the flashback return redundant?